src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1323 @@
+            if (currentRow > self->var_83C)
+                return;
+
+            int16_t currentIndustry = self->row_info[currentRow];
+            if (currentIndustry == -1)
+                return;
+
+            windows::industry::open(currentIndustry);
+        }
+
+        // 0x00458140
+        static void on_scroll_mouse_over(ui::window* self, int16_t x, int16_t y, uint8_t scroll_index)
+        {
+            self->flags &= ~(window_flags::flag_14);
+
+            uint16_t currentRow = y / common::rowHeight;
+            int16_t currentIndustry = -1;
+
+            if (currentRow < self->var_83C)
+                currentIndustry = self->row_info[currentRow];
+
+            self->row_hover = currentIndustry;
+            self->invalidate();
+        }
+
+        // 0x00457A52
+        static bool orderByName(const openloco::industry& lhs, const openloco::industry& rhs)
+        {
+            char lhsString[256] = { 0 };
+            stringmgr::format_string(lhsString, lhs.name, (void*)&lhs.town);
+
+            char rhsString[256] = { 0 };
+            stringmgr::format_string(rhsString, rhs.name, (void*)&rhs.town);
+
+            return strcmp(lhsString, rhsString) < 0;
+        }
+
+        // 0x00457A9F
+        static bool orderByStatus(openloco::industry& lhs, openloco::industry& rhs)
+        {
+            char lhsString[256] = { 0 };
+            const char* lhsBuffer = stringmgr::get_string(string_ids::buffer_1250);
+            lhs.getStatusString((char*)lhsBuffer);
+
+            stringmgr::format_string(lhsString, string_ids::buffer_1250);
+
+            char rhsString[256] = { 0 };
+            const char* rhsBuffer = stringmgr::get_string(string_ids::buffer_1250);
+            rhs.getStatusString((char*)rhsBuffer);
+
+            stringmgr::format_string(rhsString, string_ids::buffer_1250);
+
+            return strcmp(lhsString, rhsString) < 0;
+        }
+
+        // 0x00457AF3
+        static bool orderByProduction(const openloco::industry& lhs, const openloco::industry& rhs)
+        {
+            auto industryObj = objectmgr::get<industry_object>(lhs.object_id);
+            auto lhsVar = 0;
+
+            if ((industryObj->produced_cargo_type[0] & industryObj->produced_cargo_type[1]) != 0xFF)
+            {
+                lhsVar = lhs.produced_cargo_transported[0];
+                if (industryObj->produced_cargo_type[1] != 0xFF)
+                {
+                    lhsVar = lhs.produced_cargo_transported[1];
+                    if (industryObj->produced_cargo_type[0] != 0xFF)
+                    {
+                        lhsVar += lhs.produced_cargo_transported[0];
+                        lhsVar = lhsVar >> 1;
+                    }
+                }
+            }
+
+            industryObj = objectmgr::get<industry_object>(rhs.object_id);
+            auto rhsVar = 0;
+
+            if ((industryObj->produced_cargo_type[0] & industryObj->produced_cargo_type[1]) != 0xFF)
```

Comment:
Can you yank this out into a function that gets the average transported produced cargo.